src/plugins/formatters.commerce.ts

```diff
@@ -138,10 +141,191 @@ export class ProductCheckoutFormatter extends Formatter {
   }
 }
 
+type ProductPriceTemplateData = {
+  fromText?: string;
+  formattedFromPrice?: string;
+  formattedSalePriceText?: string;
+  formattedSalePrice?: string;
+  formattedNormalPriceText?: string;
+  formattedNormalPrice?: string;
+  billingPeriodValue?: number;
+  duration?: number;
+};
+
 export class ProductPriceFormatter extends Formatter {
+  private static BILLING_PERIOD_MONTHLY = 'MONTH';
```

Comment:
This was ported from the [ProductPriceFormatter](https://github.com/Squarespace/template-compiler/blob/main/core/src/main/java/com/squarespace/template/plugins/platform/CommerceFormatters.java/#L318-L471) in template-compiler